File: weddinggallery/src/main/java/com/weddinggallery/model/Photo.java
package com.weddinggallery.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "photos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Photo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "file_name", nullable = false)
    private String fileName;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploader_id")
    private User uploader;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "device_id")
    private Device device;

    @Column(name = "description")
    private String description;

    @Column(name = "upload_time", nullable = false)
    private LocalDateTime uploadTime;

    @Column(name = "comment_count", nullable = false)
    private int commentCount;

    @Column(name = "reaction_count", nullable = false)
    private int reactionCount;

    @Column(name = "visible", nullable = false)
    @Builder.Default
    private boolean visible = true;
}

File: frontend/src/types/photo.ts
export interface PhotoResponse {
  id: number;
  fileName: string;
  description: string | null;
  uploadTime: string;
  commentCount: number;
  reactionCount: number;
  uploaderUsername: string;
  deviceId: number;
  deviceName: string;
  visible: boolean;
  isVisibleForGuest: boolean;
  isWish: boolean;
  isVideo: boolean;
}

export interface PhotoDescriptionUpdateRequest {
  description: string;
}

export interface PhotoVisibilityUpdateRequest {
  visible: boolean;
}

export interface PhotoGuestVisibilityUpdateRequest {
  isVisibleForGuest: boolean;
}
